server/checks/Approval.js

```diff
@@ -209,63 +208,93 @@ export default class Approval extends Check {
+                                      // get comments that match specified approval pattern
+                                      // TODO add unicode flag once available in node
+                                      .filter(entry => {
+                                        const text = entry.body.trim()
+                                        if (entry.state) {
+                                          // `state` is available only for 'reviews' and is not for 'comments'
+                                          return entry.state === 'APPROVED'
+                                        } else {
+                                          const include = (new RegExp(approvalPattern)).test(text)
+                                          if (!include) {
+                                            info('%s: %s\'s comment "%s" does not match pattern "%s".', fullName, entry.user, text.substring(0, 16), approvalPattern)
+                                          }
+                                          return include
+                                        }
+                                      })
+                                      // reverse the order to preserve newest entries
+                                      .reverse()
+                                      // kicking out multiple approvals from same person
+                                      .filter(containsAlreadyEntryByUser)
+
+    const potentialVetos = vetoPattern ?
+      entries.filter(entry => {
+                if (entry.state) {
+                  // `state` is available only for 'reviews' and is not for 'comments'
+                  return entry.state === 'CHANGES_REQUESTED'
+                } else {
+                  const text = entry.body.trim()
+                  const include = (new RegExp(vetoPattern)).test(text)
+                  return include
+                }
+              })
+              // reverse the order to preserve newest entries
+              .reverse()
+              // kicking out multiple vetos from same person
+              .filter(containsAlreadyEntryByUser) : []
+
+    /**
+     * At this point we want to filter `potentialVetos` and `potentialApprovals`
+     * once more, based on selection in them. Approval can filter +1 out, as well as code change
+     * requirement should drop approvals
+     */
+    
+    const approvalsToUse = potentialApprovals.filter(a => {
+      const approvalTime = a.submitted_at || a.created_at
+      return potentialVetos.every(v => {
+        const vetoTime = v.submitted_at || v.created_at
+        return new Date(vetoTime) < new Date(approvalTime) || a.user !== v.user
+      })
+    })
+
+    const vetosToUse = potentialVetos.filter(v => {
+      const vetoTime = v.submitted_at || v.created_at
+      return potentialApprovals.every(a => {
+        const approvalTime = a.submitted_at || a.created_at
+        return new Date(vetoTime) > new Date(approvalTime) || a.user !== v.user
+      })
+    })
```

Comment:
This is probably not the most efficient way to filter but IMO rather explicit. As the number of comments wouldn't be any large, these operations would be quick. This can be extracted to a separate function on the one hand, on the other it would be slightly weird to pass comparison operation inside, while these wouldn't be reused anywhere else. Might be a good idea to extract though, to to unit test later on.

This part also fixes (what I think is) a bug, when a user leaves`-1` and then adds `+1` build would be blocked unless new commit is pushed.